### PR TITLE
Content provider

### LINKED ISSUE #32: Create Content Provider for Moeda
this task is to create a Content Provider to storage the data of Moeda.

### PR DESCRIPTION
**Pull request by tabocu**
Close #32.

### CODE REVIEW

File: app/src/main/java/sistemas/puc/com/finantialapp/data/FinantialContract.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+        public static final String CONTENT_TYPE =
+                ContentResolver.CURSOR_DIR_BASE_TYPE + "/" + CONTENT_AUTHORITY + "/" + PATH_MOEDA;
+        public static final String CONTENT_ITEM_TYPE =
+                ContentResolver.CURSOR_ITEM_BASE_TYPE + "/" + CONTENT_AUTHORITY + "/" + PATH_MOEDA;
+
+        public static final String TABLE_NAME = "moeda";
+
+        // TODO: Add moeda columns definitions
+        public static final String COLUMN_MOEDA_INITIALS  = "moeda_initials";
```
Comment (DanMargera) [inline app/src/main/java/sistemas/puc/com/finantialapp/data/FinantialContract.java]: `public static final String COLUMN_MOEDA_INITIALS  = "moeda_initials";`
The "initials" are called currency code. https://en.wikipedia.org/wiki/ISO_4217
Maybe rename it to COLUMN_MOEDA_CODE ?

File: app/src/main/java/sistemas/puc/com/finantialapp/data/FinantialProvider.java [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+                }
+                getContext().getContentResolver().notifyChange(uri, null);
+                return returnCount;
+            default:
+                return super.bulkInsert(uri, values);
+        }
+    }
+
+    @Override
+    @TargetApi(11)
```
Comment (DanMargera) [inline app/src/main/java/sistemas/puc/com/finantialapp/data/FinantialProvider.java:258]: `@TargetApi(11)`
I'm not sure but maybe this should be a constant somewhere?

File: app/src/main/java/sistemas/puc/com/finantialapp/data/FinantialProvider.java [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+        }
+        // Because a null deletes all rows
+        if (rowsDeleted != 0) {
+            getContext().getContentResolver().notifyChange(uri, null);
+        }
+        return rowsDeleted;
+    }
+
+    @Override
+    public int update(@NonNull Uri uri, @Nullable ContentValues values, @Nullable String selection, @Nullable String[] selectionArgs) {
```
Comment (DanMargera) [inline app/src/main/java/sistemas/puc/com/finantialapp/data/FinantialProvider.java]: This line is long enough to break.

File: app/src/main/java/sistemas/puc/com/finantialapp/data/FinantialProvider.java [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+                return FinantialContract.TesouroEntry.CONTENT_TYPE;
+            case INDICE:
+                return FinantialContract.IndiceEntry.CONTENT_TYPE;
+            default:
+                throw new UnsupportedOperationException("Unknown uri: " + uri);
+        }
+    }
+
+    @Override
+    public Cursor query(@NonNull Uri uri, @Nullable String[] projection, @Nullable String selection, @Nullable String[] selectionArgs, @Nullable String sortOrder) {
```
Comment (DanMargera) [inline app/src/main/java/sistemas/puc/com/finantialapp/data/FinantialProvider.java]: This line is long enough to break.

File: app/src/main/java/sistemas/puc/com/finantialapp/data/FinantialProvider.java (hunk 2 above)
Comment (tabocu) [reply app/src/main/java/sistemas/puc/com/finantialapp/data/FinantialProvider.java:258]: This is used with so many API numbers. It should be clear for those who read it. Also, I never saw an example which extract this kind of information to constants.

File: app/src/main/java/sistemas/puc/com/finantialapp/data/FinantialContract.java (hunk 1 above)
Comment (tabocu) [reply app/src/main/java/sistemas/puc/com/finantialapp/data/FinantialContract.java]: I liked your idea. I was looking for a better name. Thank you.

File: app/src/main/java/sistemas/puc/com/finantialapp/data/FinantialContract.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                BASE_CONTENT_URI.buildUpon().appendPath(PATH_MOEDA).build();
+
+        public static final String CONTENT_TYPE =
+                ContentResolver.CURSOR_DIR_BASE_TYPE + "/" + CONTENT_AUTHORITY + "/" + PATH_MOEDA;
+        public static final String CONTENT_ITEM_TYPE =
+                ContentResolver.CURSOR_ITEM_BASE_TYPE + "/" + CONTENT_AUTHORITY + "/" + PATH_MOEDA;
+
+        public static final String TABLE_NAME = "moeda";
+
+        public static final String COLUMN_MOEDA_CODE = "moeda_initials";
```
Comment (DanMargera) [inline app/src/main/java/sistemas/puc/com/finantialapp/data/FinantialContract.java]: Not that it makes any difference, but change the constant value to "moeda_code" for consistency.

**Review by DanMargera — APPROVED** [review]

File: app/src/main/java/sistemas/puc/com/finantialapp/data/FinantialContract.java (hunk 5 above)
Comment (tabocu) [reply app/src/main/java/sistemas/puc/com/finantialapp/data/FinantialContract.java]: You're right! I will change it!

**Pull request merged**